generators/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/processor/response/ResponseStrategyHelper.java

```diff
@@ -41,7 +41,7 @@ Response responseFor(final String action, final Optional<JsonEnvelope> result, f
         if (result.isPresent()) {
             final JsonEnvelope outputEnvelope = result.get();
 
-            if (outputEnvelope.payload() == NULL) {
+            if (outputEnvelope.payload() == NULL || outputEnvelope.payload() == null) {
```

Comment:
Check should probably be the other way around - check for null first then NULL. Also needs a test case!